### DISCUSSION TITLE
Debug command for overriding VGA display start address

### DISCUSSION THREAD

**Discussion by pppq** [joncampbell123/dosbox-x#4715] [Q&A] [answered]
Hi all,

I'm looking for a way to display a different section of memory as pixel data (a game's back buffer), similar to request #1318. 

The `VGA DS START <addr>` debug command that appeared in the 2023-10 build looks like it would work for this purpose, however the only effect I see is the previously displayed contents of VGA memory getting shifted around in the window.

Example debug session:
```
> VGA DS X

> VGA MODE
VGA mode info:
mode=M_VGA vref=70.087Hz href=31468.750Hz chrclk=3146875.000Hz dotclk=2517500...
disp-start=0h real-start=0h retrace=0 scanlen=40 cursor-start=0h
[...]

> VGA DRAW
VGA draw info:
resizing=0 width=320 height=200 blocks=80
address=3e800h panning=0 bytes-skip=0 lin-mask=3ffff pln-mask=ffff
addr-add=500h line-length=1280 addrline-total=2 addrline=0
line-total=200 vblank-skip=0 lines-done=200 split-line=512
[...]

> VGA DS START 10000

> VGA MODE
VGA mode info:
mode=M_VGA vref=70.087Hz href=31468.750Hz chrclk=3146875.000Hz dotclk=2517500...
disp-start=0h real-start=10000h retrace=0 scanlen=40 cursor-start=0h
[...]

> VGA DRAW
VGA draw info:
resizing=0 width=320 height=200 blocks=80
address=13e800h panning=0 bytes-skip=0 lin-mask=3ffff pln-mask=ffff
addr-add=500h line-length=1280 addrline-total=2 addrline=0
line-total=200 vblank-skip=0 lines-done=200 split-line=512
[...]
```

I dug a bit into how this might work but didn't get very far unfortunately 😅 Can the command be used for this purpose at all? What type of address (eg. virtual, physical, real-mode segment) does it expect?

**Comment by joncampbell123** [comment, marked as answer]
The command affects the display start address (within video memory) in the CRTC registers, not the memory address that VGA memory is mapped as seen by the CPU.
It can be used for example to peer at off-screen portions of video memory for debugging purposes.

**Reply by pppq** [reply]
Thanks!
